src/klein/_isession.py

```diff
@@ -328,15 +320,33 @@ def finalize():
         """
 
 
+class _IRequestLifecycle(Interface):
+    """
+    Interface for adding hooks to the phases of a request's lifecycle.
+    """
+
+
+if TYPE_CHECKING:
+    from typing import Union
+    from ._requirer import RequestLifecycle
+
+    IRequestLifecycle = Union[_IRequestLifecycle, RequestLifecycle]
+else:
+    IRequestLifecycle = _IRequestLifecycle
```

Comment:
When type checking, we use a `Union` to work around lack of support for `zone.interface` in Mypy.

Normally we do this in `interfaces.py`, but we're doing it here (also) due to the fact that we use `IRequestLifecycle` later in this file and importing from `interfaces.py` will cause a circular import loop.

All of this should go away when we start using the Zope plug-in for Mypy.